src/drivers/cmakeDriver.ts

```diff
@@ -485,78 +485,113 @@ export abstract class CMakeDriver implements vscode.Disposable {
      * Change the current configure preset. This lets the driver reload, if necessary.
      * @param configurePreset The new configure preset
      */
-    async setConfigurePreset(configurePreset: preset.ConfigurePreset): Promise<void> {
-        if (!this.useCMakePresets) {
-            log.info(localize('skip.set.config.preset', 'Using kits, skip setting configure preset: {0}', configurePreset.name));
-            return;
-        }
+    async setConfigurePreset(configurePreset: preset.ConfigurePreset | null): Promise<void> {
+        if (configurePreset) {
+            if (!this.useCMakePresets) {
+                log.info(localize('skip.set.config.preset', 'Using kits, skip setting configure preset: {0}', configurePreset.name));
+                return;
+            }
 
-        log.info(localize('switching.to.config.preset', 'Switching to configure preset: {0}', configurePreset.name));
+            log.info(localize('switching.to.config.preset', 'Switching to configure preset: {0}', configurePreset.name));
 
-        const newBinaryDir = configurePreset.binaryDir;
-        const needs_clean = this.binaryDir === newBinaryDir && preset.configurePresetChangeNeedsClean(configurePreset, this._configurePreset);
-        await this.doSetConfigurePreset(needs_clean, async () => {
-            await this._setConfigurePreset(configurePreset);
-        });
+            const newBinaryDir = configurePreset.binaryDir;
+            const needs_clean = this.binaryDir === newBinaryDir && preset.configurePresetChangeNeedsClean(configurePreset, this._configurePreset);
+            await this.doSetConfigurePreset(needs_clean, async () => {
+                await this._setConfigurePreset(configurePreset);
+            });
+        } else {
+            if (!this.useCMakePresets) {
+                log.info(localize('skip.unset.config.preset', 'Using kits, skip unsetting configure preset'));
```

Comment:
Is this called in real scenarios?  It seems like a bug if we're trying to set a preset while using kits.

Reply:
The original code has this, so I duplicated it. Technically, I guess this can probably happen when user invokes the command directly, bypassing the UI?

Reply:
I suppose another extension could theoretically execute the command even though it's disabled in the UI, but I would think we should just exit early as opposed to logging this behavior.